koku/api/user_access/view.py

```diff
@@ -28,121 +28,88 @@ class UIFeatureAccess:
+            if query_param in self.access_key_mapping:
+                self.access_key_list = [query_param]
+        else:
+            self.access_key_list = self.access_key_mapping.keys()
+        return bool(self.access_key_list)
+
+    @property
+    def access_key_mapping(self):
+        mapping = {}
+        mapping["aws"] = ["aws.account", "aws.organizational_unit"]
+        mapping["ocp"] = ["openshift.cluster", "openshift.node", "openshift.project"]
+        mapping["azure"] = ["azure.subscription_guid"]
+        mapping["gcp"] = ["gcp.account", "gcp.project"]
+        mapping["oci"] = ["oci.payer_tenant_id"]
+        mapping["ibm"] = ["ibm.account"]
+        mapping["cost_model"] = ["cost_model"]
+        mapping["settings"] = ["settings"]
+        mapping["any"] = mapping["aws"] + mapping["azure"] + mapping["gcp"] + mapping["oci"] + mapping["ocp"]
+        return mapping
 
     def _get_access_value(self, key1, key2, default=None):
         """Return the access value from the inner dict."""
         return self.access_dict.get(key1, {}).get(key2, default)
 
-    def access(self, admin_user, pre_release_feature):
+    def _check_access(self, pre_release_feature, access_key):
         """Access property returns whether the user has the requested access.
 
         Return:
-            (bool)
+            (access, read_only)
+            acesss = bool or None
+            read_only = bool or none
         """
-        if pre_release_feature and not settings.ENABLE_PRERELEASE_FEATURES:
-            return False
-
-        if admin_user:
-            return True
-
-        for key in self.access_keys:
-            if self._get_access_value(key, "read") or self._get_access_value(key, "write"):
-                return True
-        return False
-
-
-class AWSUserAccess(UIFeatureAccess):
-    """Access to AWS UI Features."""
-
-    access_keys = ["aws.account", "aws.organizational_unit"]
-
-
-class OCPUserAccess(UIFeatureAccess):
-    """Access to OCP UI Features."""
-
-    access_keys = ["openshift.cluster", "openshift.node", "openshift.project"]
-
-
-class AzureUserAccess(UIFeatureAccess):
-    """Access to Azure UI Features."""
-
-    access_keys = ["azure.subscription_guid"]
-
-
-class GCPUserAccess(UIFeatureAccess):
-    """Access to GCP UI Features."""
-
-    access_keys = ["gcp.account", "gcp.project"]
 
+        if pre_release_feature and not settings.ENABLE_PRERELEASE_FEATURES:
+            return False, None
+        if self.admin_user:
+            return True, False
+        for rbac_setting in self.access_key_mapping[access_key]:
+            if self._get_access_value(rbac_setting, "write"):
+                return True, False
+            elif self._get_access_value(rbac_setting, "read"):
+                return True, True
+        return False, None
+
+    def generate_data(self):
+        data = []
+        for access_key in self.access_key_list:
+            pre_release_feature = bool(access_key in self.PRERELEASE)
```

Comment:
```suggestion
            pre_release_feature = access_key in self.PRERELEASE
```
This type-cast is redundant.